iohelper/iohelper.go

```diff
@@ -0,0 +1,95 @@
+package iohelper
+
+/*
+ * This file contains structs and functions related to interacting with files
+ * and directories.
+ */
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/greenplum-db/gp-common-go-libs/gplog"
+	"github.com/greenplum-db/gp-common-go-libs/operating"
+	"github.com/pkg/errors"
+)
+
+/*
+ * Generic file/directory manipulation functions that use the function pointers
+ * and types from the operating package, for easier test mocking around file IO.
+ */
+
+func OpenFileForReading(filename string) (operating.ReadCloserAt, error) {
+	fileHandle, err := operating.System.OpenFileRead(filename, os.O_RDONLY, 0644)
+	if err != nil {
+		return nil, errors.Errorf("Unable to open file for reading: %s", err)
+	}
+	return fileHandle, nil
+}
+
+func MustOpenFileForReading(filename string) operating.ReadCloserAt {
+	fileHandle, err := OpenFileForReading(filename)
+	gplog.FatalOnError(err)
+	return fileHandle
+}
+
+func OpenFileForWriting(filename string, allowAppend ...bool) (io.WriteCloser, error) {
+	flags := os.O_CREATE | os.O_WRONLY
+	if len(allowAppend) == 1 && allowAppend[0] {
+		flags = os.O_APPEND | flags
+	}
+	fileHandle, err := operating.System.OpenFileWrite(filename, flags, 0644)
+	if err != nil {
+		return nil, errors.Errorf("Unable to create or open file for writing: %s", err)
+	}
+	return fileHandle, nil
+}
+
+func MustOpenFileForWriting(filename string, allowAppend ...bool) io.WriteCloser {
+	fileHandle, err := OpenFileForWriting(filename, allowAppend...)
+	gplog.FatalOnError(err)
+	return fileHandle
+}
+
+func FileExistsAndIsReadable(filename string) bool {
```

Comment:
We should pass through the error, otherwise we don't know whether the file didn't exist or couldn't be read due to permissions.

Reply:
I'm not sure we would be parsing the error where this is used. Seems like it might just be "If FileExistsAndIsReadable do x else do y"

Reply:
Correct, we don't care what the error is in this case, just that it's not currently usable.